sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -245,25 +282,26 @@ object GpuDeviceManager extends Logging {
         } else {
           conf.rmmAllocReserve
         }
+      val effectiveFree = Math.min(info.free, effectiveGpuTotal)
       var poolAllocation = truncateToAlignment(
-        (conf.rmmAllocFraction * (info.free - reserveAmount)).toLong)
+        (conf.rmmAllocFraction * (effectiveFree - reserveAmount)).toLong)
       val errorPhrase = "The pool allocation of " +
-        s"${toMiB(poolAllocation)} MiB (gpu.free: ${toMiB(info.free)}," +
+        s"${toMiB(poolAllocation)} MiB (effective free: ${toMiB(effectiveFree)}," +
```

Comment:
as a use I don't necessarily know what `effective` is. We are also not consistently using the `effective` name in the logs. We use `gpu.total` for really `total effective`. I think this is getting into the weeds a bit. I would just use the old labels for these variables. I would add a log line when you compute `effectiveGpuTotal` however at startup, no matter what. That way we know what the "GPU" memory we are working with is.